src/components/TypesOfRecipeCards/RecipeCard.jsx

```diff
@@ -0,0 +1,82 @@
+import { PushPin } from "@phosphor-icons/react";
+import { Link } from "react-router-dom";
+import { pinWhite, pinSize } from "../../shared/style";
+
+const RecipeCard = ({ pinned, recipe }) => {
```

Comment:
```suggestion
const ProfilePost = ({ pinned, post }) => {
  isRecipe = ...
```
then use the boolean to conditionally render